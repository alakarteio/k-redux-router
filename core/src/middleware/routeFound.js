const type = '@@router/ROUTE_FOUND'

export default Object.assign(
  result => ({ type, payload: Object.assign({ found: true }, result) }),
  {
    type,
  },
)
