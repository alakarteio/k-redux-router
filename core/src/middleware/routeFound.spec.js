/* eslint-env jest */
import routeFound from './routeFound'

describe('routeFound', () => {
  const match = res => () => expect(res).toMatchSnapshot()
  it('should returns the type', match(routeFound.type))
  it('should returns found: true', match(routeFound({ result: 'payload' })))
  it('should returns found: false (overrided)', match(routeFound({ result: 'payload', found: false })))
})
