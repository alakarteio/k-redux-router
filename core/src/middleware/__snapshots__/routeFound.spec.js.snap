// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routeFound should returns found: false (overrided) 1`] = `
Object {
  "payload": Object {
    "found": false,
    "result": "payload",
  },
  "type": "@@router/ROUTE_FOUND",
}
`;

exports[`routeFound should returns found: true 1`] = `
Object {
  "payload": Object {
    "found": true,
    "result": "payload",
  },
  "type": "@@router/ROUTE_FOUND",
}
`;

exports[`routeFound should returns the type 1`] = `"@@router/ROUTE_FOUND"`;
